File: careForEldersBack/event/src/main/java/com/care4elders/event/entity/Event.java
package com.care4elders.event.entity;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Document(collection = "events")
@Data
public class Event {
    @Id
    private String id;
    private String title;
    private Date date;
    private String location;
    private String description;
    private List<String> participantIds; // Stores user IDs from user-service
}

File: careForEldersFront/src/app/models/Event.ts
// models/Event.ts
export interface Event {
  target: HTMLInputElement;
  id?: string;
  title: string;
  date: Date | string;
  location: string;
  description: string;
  participantIds: string[];
}
export interface EventDTO {
  title: string;
  date: Date;
  location: string;
  description: string;
  participantIds: string[];
}
export interface User {
  id: string;
  name: string;
 role: string;
}